client/src/PublicRequest.ts

```diff
@@ -187,32 +187,52 @@ export type SignBtcTransactionRequest
 
 export type SignSwapRequest = SimpleRequest & {
     swapId: string,
-    fund:
+    fund: (
         {type: 'NIM'}
-            & Omit<TransactionInfo,
-                'senderType'
-                | 'recipient'
-                | 'recipientType'
-                | 'recipientLabel'
-                | 'validityStartHeight'
-                | 'data'
-                | 'flags'
-            >
-            & { senderLabel: string }
-        | {type: 'BTC', keyPaths: string[], value: number, fee: number },
-    redeem:
+        & Omit<TransactionInfo,
```

Comment:
Using `Pick` instead of `Omit` would probably be shorter and give a better idea of what's actually included.

Reply:
The type has 11 properties, so `Pick` would be longer.

It would become this:
```
        & Pick<TransactionInfo,
            | 'keyPath'
            | 'senderLabel'
            | 'sender'
            | 'senderType'
            | 'value'
            | 'fee'
            | 'validityStartHeight'
        >
```

But it would loose all the explanation of why the other ones are omitted. I prefer the `Omit<>` method.